### PR TITLE
Azure Arc-enabled servers Private Link integration

### PR DESCRIPTION
**Pull request by lanicolas**
Added scenario for private connectivity from an Azure Arc enabled server to Azure SQL database via VPN and private link

### CODE REVIEW

File: docs/azure_arc_jumpstart/azure_arc_servers/day2/arc_privatelink/_index.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+linkTitle: "Azure Private Link"
+weight: 11
+description: >
+---
+
+## Use Azure Private Link to securely connect networks to Azure Arc
+
+The following README will guide you on how to use [Azure Private Link](https://docs.microsoft.com/en-us/azure/private-link/private-link-overview) to securely connect from an Azure Arc-enabled server to Azure using a VPN. [This feature](https://docs.microsoft.com/en-us/azure/azure-arc/servers/private-link-security) allows you to connect privately to Azure Arc without opening up public network access but rather using private endpoints over a VPN or ExpressRoute connection, ensuring that all traffic is being sent to Azure privately.
+
+In this guide, you will emulate a hybrid environment connected to Azure over a VPN with hybrid resources that will be Arc-enabled, and Azure Private Link Scope will be used to connect over a private connection. To complete this process you deploy a single ARM template that will:
```
Comment (dkirby-ms) [inline docs/azure_arc_jumpstart/azure_arc_servers/day2/arc_privatelink/_index.md:13]:
```suggestion
In this guide, you will emulate a hybrid environment connected to Azure over a VPN with hybrid resources that will be Azure Arc-enabled, and Azure Private Link Scope will be used to connect over a private connection. To complete this process you deploy a single ARM template that will:
```

File: docs/azure_arc_jumpstart/azure_arc_servers/day2/arc_privatelink/_index.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+- Create two separate resource groups:
+
+  - "On-premises" resource group: will simulate a private on-premises environment with a Windows virtual machine. This VM will not have a public IP address assigned to it so [Azure Bastion](https://docs.microsoft.com/en-us/azure/bastion/bastion-overview) is deployed to have administrative access to the operating system. The Windows virtual machine is an Azure Arc-enabled server by installing the Azure Arc-connected machine agent using Azure Private Link.
```
Comment (dkirby-ms) [inline docs/azure_arc_jumpstart/azure_arc_servers/day2/arc_privatelink/_index.md:17]:
```suggestion
  - "On-premises" resource group will contain resources that simulate a private on-premises environment with a Windows virtual machine. This VM will not have a public IP address assigned to it so [Azure Bastion](https://docs.microsoft.com/en-us/azure/bastion/bastion-overview) is deployed to have administrative access to the operating system. The Windows virtual machine is an Azure Arc-enabled server by installing the Azure Arc connected machine agent using Azure Private Link.
```

File: docs/azure_arc_jumpstart/azure_arc_servers/day2/arc_privatelink/_index.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  - "Azure" resource group: in this resource group, you will have the Azure Arc-enabled server and its Private Link Scope.
```
Comment (dkirby-ms) [inline docs/azure_arc_jumpstart/azure_arc_servers/day2/arc_privatelink/_index.md:18]:
```suggestion
  - "Azure" resource group will contain the Azure Arc-enabled server and its Private Link Scope.
```

File: docs/azure_arc_jumpstart/azure_arc_servers/day2/arc_privatelink/_index.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+- Both resource groups have their own virtual networks and address spaces, and they are connected via Azure VPN gateways to set up a hybrid private connection.
+
+  ![Deployment Overview](./01.png)
+
+  > **NOTE: It is not expected for an Azure VM to be projected as an Azure Arc-enabled server. The below scenario is unsupported and should ONLY be used for demo and testing purposes.**
+  > **NOTE: The below scenario assumes the on-premises VM has outbound internet connectivity for the deployment of the Azure Arc connected machine agent, for internet disconnected environments you will need to adjust the automation to retrieve the agent's software from locally accessible storage**
```
Comment (dkirby-ms) [inline docs/azure_arc_jumpstart/azure_arc_servers/day2/arc_privatelink/_index.md:25]:
```suggestion
  > **NOTE: The below scenario assumes the on-premises VM has outbound internet connectivity for the deployment of the Azure Arc connected machine agent. For internet disconnected environments you will need to adjust the automation to retrieve the agent's software from locally accessible storage**
```

**Review by dkirby-ms — CHANGES_REQUESTED** [review]
Please see my comments and address the following. It might also be good to have a more descriptive explanation on why we are using Bastion.

* Switch from using c:\tmp folder to c:\Temp to align with other scenarios
* Azure Arc-enabled branding
* Use on-premises instead of onpremise
* Minor grammar fixes
* Minor changes to align with other JS scenarios/guidelines

File: azure_arc_servers_jumpstart/privatelink/artifacts/bootstrap.ps1 [hunk 5]
```diff
@@ -0,0 +1,68 @@
+param (
```
[the remark this answers is not in the document]
Comment (lanicolas) [reply azure_arc_servers_jumpstart/privatelink/artifacts/bootstrap.ps1:1]: changed

File: azure_arc_servers_jumpstart/privatelink/artifacts/bootstrap.ps1 [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+[System.Environment]::SetEnvironmentVariable('tenantId', $tenantId,[System.EnvironmentVariableTarget]::Machine)
+[System.Environment]::SetEnvironmentVariable('resourceGroup', $resourceGroup,[System.EnvironmentVariableTarget]::Machine)
+[System.Environment]::SetEnvironmentVariable('subscriptionId', $subscriptionId,[System.EnvironmentVariableTarget]::Machine)
+[System.Environment]::SetEnvironmentVariable('Location', $location,[System.EnvironmentVariableTarget]::Machine)
+[System.Environment]::SetEnvironmentVariable('PEname', $PEname,[System.EnvironmentVariableTarget]::Machine)
+[System.Environment]::SetEnvironmentVariable('adminUsername', $adminUsername,[System.EnvironmentVariableTarget]::Machine)
+[System.Environment]::SetEnvironmentVariable('PLscope', $PLscope,[System.EnvironmentVariableTarget]::Machine)
+
+# Creating Log File
+New-Item -Path "C:\" -Name "tmp" -ItemType "directory" -Force
```
[the remark this answers is not in the document]
Comment (lanicolas) [reply azure_arc_servers_jumpstart/privatelink/artifacts/bootstrap.ps1:24]: changed

File: azure_arc_servers_jumpstart/privatelink/artifacts/bootstrap.ps1 [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+Start-Transcript -Path C:\tmp\LogonScript.log
```
[the remark this answers is not in the document]
Comment (lanicolas) [reply azure_arc_servers_jumpstart/privatelink/artifacts/bootstrap.ps1:25]: fixed thanks

File: azure_arc_servers_jumpstart/privatelink/nesteddeploy.example.parameters.json [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+            "value": "Standard_D4s_v4"
+        },
+        "adminUsername": {
+            "value": "arcdemo"
+        },
+        "adminPassword": {
+            "value": "ArcPassword123!!"
+        },
+        "onPremVNetName": {
+            "value": "OnPremise-VNET"
```
[the remark this answers is not in the document]
Comment (lanicolas) [reply azure_arc_servers_jumpstart/privatelink/nesteddeploy.example.parameters.json:48]: changed

File: azure_arc_servers_jumpstart/privatelink/nesteddeploy.json [hunk 9]
```diff
@@ -0,0 +1,541 @@
+{
```
[the remark this answers is not in the document]
Comment (lanicolas) [reply azure_arc_servers_jumpstart/privatelink/nesteddeploy.json:1]: any other idea?

File: azure_arc_servers_jumpstart/privatelink/ARM/onpremisedeploy.json [hunk 10]
```diff
@@ -0,0 +1,311 @@
+{
```
[the remark this answers is not in the document]
Comment (lanicolas) [reply azure_arc_servers_jumpstart/privatelink/ARM/onpremisedeploy.json:1]: this deploys the on-premises part, so not sure how other way to call it.

File: azure_arc_servers_jumpstart/privatelink/nesteddeploy.parameters.json [hunk 11]
```diff
@@ -0,0 +1,78 @@
+{
```
[the remark this answers is not in the document]
Comment (lanicolas) [reply azure_arc_servers_jumpstart/privatelink/nesteddeploy.parameters.json:1]: if there are any ideas for file names I can change for sure

File: azure_arc_servers_jumpstart/privatelink/nesteddeploy.example.parameters.json [hunk 12]
```diff
@@ -0,0 +1,84 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "azureRgName": {
+            "value": "Arc-Azure-PL-demo1"
+        },
+        "onPremRgName": {
+            "value": "Arc-Onpremise-PL-Demo1"
```
[the remark this answers is not in the document]
Comment (lanicolas) [reply azure_arc_servers_jumpstart/privatelink/nesteddeploy.example.parameters.json]: changed

**Comment by lanicolas** [comment]
discussed with Lior and agreed on new file names as suggested by Dale, all should be ready now

File: azure_arc_servers_jumpstart/privatelink/nesteddeploy.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (lanicolas) [reply azure_arc_servers_jumpstart/privatelink/nesteddeploy.json]: changed all file names

**Comment by lanicolas** [comment]
just added the latest diagram

**Review by likamrat — APPROVED** [review]
LGTM

**Pull request merged**